src/main/java/jdbchomework/dao/hibernate/AbstractHibDao.java

```diff
@@ -71,49 +75,48 @@ public T getById(long id) {
                 transaction.rollback();
             }
             log.error(CANNOT_CONNECT_TO_DB, e);
+            throw new RuntimeException(CANNOT_CONNECT_TO_DB, e);
         }
         return result;
     }
 
-    public boolean updateById(long id, T toUpdate) {
-        boolean result = false;
-        Session session = getSession();
+    @Override
+    public boolean deleteById(long id) {
         Transaction transaction = null;
-        try {
+        boolean res = false;
+        try (Session session = sessionFactory.openSession()) {
             transaction = session.beginTransaction();
-            toUpdate.setId(id);
-            session.update(toUpdate);
+            T t = getById(id);
+            session.delete(t);
             transaction.commit();
+            res = true;
         } catch (HibernateException e) {
             if (transaction != null) {
                 transaction.rollback();
             }
             log.error(CANNOT_CONNECT_TO_DB, e);
+            throw new RuntimeException(CANNOT_CONNECT_TO_DB, e);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS00112)